crates/bevy_ecs/src/schedule/runner/mod.rs

```diff
@@ -0,0 +1,190 @@
+mod multi_threaded;
+mod single_threaded;
+
+use crate::{
+    schedule::{BoxedRunCondition, RegistryId, SystemLabel, SystemRegistry},
+    system::BoxedSystem,
+    world::World,
+};
+
+use self::single_threaded::{SimpleRunner, SingleThreadedRunner};
+
+#[cfg(feature = "trace")]
+use bevy_utils::tracing::Instrument;
+use bevy_utils::HashMap;
+
+use downcast_rs::{impl_downcast, Downcast};
+use fixedbitset::FixedBitSet;
+
+/// Types that can run [`System`] instances on the data stored in a [`World`].
+pub trait SystemRunner: Downcast + Send + Sync {
+    fn run(&mut self, world: &mut World);
+}
+
+impl_downcast!(SystemRunner);
+
+/// Internal resource used to signal the runner to apply pending commands to the [`World`].
+pub(super) struct RunnerApplyBuffers(pub bool);
+
+impl Default for RunnerApplyBuffers {
+    fn default() -> Self {
+        Self(false)
+    }
+}
+
+/// Applies pending command queues to the [`World`].
+pub fn apply_buffers(world: &mut World) {
+    // Best place to do this check.
+    world.check_change_ticks();
+
+    let mut should_apply_buffers = world
+        .get_resource_mut::<RunnerApplyBuffers>()
```

Comment:
This should be `.resource_mut()`. I wrote a nice generic error message for this!